RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -0,0 +1,148 @@
+    [InlineData(2, StatusCodes.Status200OK)]
+    [InlineData(3, StatusCodes.Status200OK)]
+    [InlineData(4, StatusCodes.Status503ServiceUnavailable)]
+    [InlineData(5, StatusCodes.Status200OK)]
+    public async Task GetBrewStatus_WhenEndpointCalled_ShouldReturnCorrectStatusCode(long numRequestsSoFar, int expectedStatusCode)
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+        });
+
+        response.ShouldNotBeNull();
+        ((int)response.StatusCode).ShouldBe(expectedStatusCode);
+    }
+
+    [Theory]
+    [InlineData(0)]
+    [InlineData(4)]
+    public async Task GetBrewStatus_WhenApril1st_ShouldReturnEmpty418(long numRequestsSoFar)
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+
+        var dateTimeProvider = new Mock<IDateTimeProvider>();
+        dateTimeProvider.Setup(p => p.GetLocalNow())
+            .Returns(new DateTime(2023, 4, 1));
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
+        });
+
+        ((int)response.StatusCode).ShouldBe(StatusCodes.Status418ImATeapot);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenCoffeeReady_ShouldReturn200()
+    {
+        var nowLocal = new DateTime(2023, 1, 2, 3, 4, 5, DateTimeKind.Local);
+        var dateTimeProvider = new Mock<IDateTimeProvider>();
+        dateTimeProvider.Setup(p => p.GetLocalNow())
+            .Returns(nowLocal);
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
+        var content = await response.Content.ReadFromJsonAsync<BrewStatusResponse>();
+        content.ShouldNotBeNull();
+        content.Message.ShouldBe("Your piping hot coffee is ready");
+
+        var offsetStr = DateTime.Now.ToString("zzz").Replace(":", "");
+        var expectedPrepared = $"2023-01-02T03:04:05{offsetStr}";
+        content.Prepared.ShouldBe(expectedPrepared);
+    }
+
+    private Task<HttpResponseMessage> GetEndpointResponse(Action<IServiceCollection> servicesConfiguration)
+    {
+        var mockFactory = _factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureTestServices(servicesConfiguration);
+        });
+
+        var client = mockFactory.CreateClient();
+        return client.GetAsync("brew-coffee");
```

Comment:
The primary reason for introducing a test helper method is to reduce code duplication, but it also makes future changes cheaper.

For instance, if the endpoint name were to change from `brew-coffee` to `coffee-status`, or changed from `GET` to `POST`.